src/v/config/property.h

```diff
@@ -197,6 +197,10 @@ class property : public base_property {
         return std::nullopt;
     }
 
+    bool check_restricted(const T& value) const {
+        return do_check_restricted(value);
+    }
```

Comment:
nitpick: what is the purpose of this indirection?

Reply:
it was not not introduce a virtual function in the interface of the property.
All the other functions are either overriding a `base_property` virtual or are non-virtual functions.
https://github.com/redpanda-data/redpanda/pull/23915#discussion_r1820578359

Reply:
fair enough!